lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -637,113 +654,169 @@
+    // Windows National Language Support (NLS) uses RFC4646 language tags, and NLS currently provides
+    // the backing implementation for platform.getDefaultLocale() and platform.compareString() on Windows.
+    // Windows Globalization, ICU, and the user-exposed Intl interface use RFC5646 language tags
+    // The following code should be used immediately before or after calling an NLS-backed platform method
+    const [toNLS, fromNLS] = (function () {
+        // ICU does not need to worry about NLS locale strings
+        if (isPlatformUsingICU) {
+            const identity = (locale) => locale;
+            return [identity, identity];
+        }
 
-            callInstanceFunc(ArrayInstanceForEach, ObjectGetOwnPropertyNames(collationValuesForLocale), function (collation) {
-                const windowsTag = collationValuesForLocale[collation];
-                if (windowsTag !== "") {
-                    reversedCollationValues[windowsTag] = collation;
-                }
-            });
+        // If an empty string is encountered for the value of the property; that means that is by default.
+        // So in the case of zh-TW; "default" and "stroke" are the same.
+        // This list was discussed with AnBorod, AnGlass and SureshJa.
+        // Source of this list: https://msdn.microsoft.com/en-us/library/cc233968.aspx (the table at the bottom)
+        // for each sub-object, the key is a RFC5646 collation value and the value is an RFC4646 collation value
+        const localesAcceptingCollationValues = bare({
+            "es-ES": bare({ trad: "tradnl" }),
+            "lv-LV": bare({ trad: "tradnl" }),
+            "de-DE": bare({ phonebk: "phoneb" }),
+            "ja-JP": bare({ unihan: "radstr" }),
+            "zh-TW": bare({ unihan: "radstr", stroke: "", phonetic: "pronun" }),
+            "zh-HK": bare({ unihan: "radstr", stroke: "" }),
+            "zh-MO": bare({ unihan: "radstr", stroke: "" }),
+            "zh-CN": bare({ stroke: "stroke", pinyin: "" }),
+            "zh-SG": bare({ stroke: "stroke", pinyin: "" })
+
+            // The following locales are supported by Windows; however, no BCP47 equivalent collation values were found for these.
+            // In future releases; this list (plus most of the Collator implementation) will be changed/removed as the platform support is expected to change.
+            // "hu-HU": ["technl"],
+            // "ka-GE": ["modern"],
+            // "x-IV": ["mathan"]
+        });
 
-            toReturn[locale] = reversedCollationValues;
+        // reverses the keys and values in each locale's sub-object in localesAcceptingCollationValues
+        // localesAcceptingCollationValues[locale][key] = value -> reverseLocalesAcceptingCollationValues[locale][value] = key
+        // for each sub-object, the key is a RFC4646 collation value and the value is an RFC5646 collation value
+        const reverseLocalesAcceptingCollationValues = bare({
```

Comment:
>reverseLocalesAcceptingCollationValues [](start = 14, length = 38)

did you just write this out manually? Seems like a pretty straightforward transformation of the above table. Maybe add a comment about keeping these two objects in sync?

Reply:
I could, but my understanding was that these values would _never_ change since they are defined by LCIDs underneath